controllers/usersignupcleanup/usersignup_cleanup_controller.go

```diff
@@ -83,7 +82,7 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 	}
 	reqLogger = reqLogger.WithValues("username", instance.Spec.Username)
 
-	if states.VerificationRequired(instance) && !states.Approved(instance) {
+	if states.VerificationRequired(instance) && !condition.IsTrue(instance.Status.Conditions, toolchainv1alpha1.UserSignupApproved) {
```

Comment:
While this will work I'm concert with the approach of relaying on leftover status conditions here...
Ideally we should introduce a new condition or maybe a new state? Something like `previously-provisioned` or something like this to mark signups which were at least once approved/provisioned in the past.
Just to separate the state/status of `approved-at-least-once-in-the-past` from `currently-approved`.
But it would require much bigger changes in the code. So, let's keep it simple for now and think about improving later when we have time.

But let's add a very detailed comment here in the code with explaining why we are checking the Approved status here.

Reply:
yeah, it's maybe not the best, but it was the quickest solution :stuck_out_tongue_closed_eyes: 

Another solution (without adding a new condition/status) would be checking the `toolchain.dev.openshift.com/activation-counter` annotations if it contains a non-zero value. But in this case I had a concern that we use the annotation only for the metrics, and it doesn't have anything in common with cleanup or any kind of state/status.

Reply:
better? d95ef07

Reply:
Actually.. now I remember. I thought about using this annotation in the first place. I think it's not that bad approach. This annotation is exactly for counting activations and this is what we need. We don't want to cleanup such resources because of the metrics. So, I think it's fine to use the annotation and we don't need any new status/state.
WDYT?

Reply:
well, both approaches have pros and cons... Using the annotation was my first thought, but since we mainly rely on conditions in that controller, I decided to use the same approach. The second reason for using the condition was what I described in the previous comment.

But I'm fine with using the annotation too. I can change it, but not today - I can do it on Monday. Or - if you want - you can contribute to my PR :wink:

Reply:
I'm on PTO today ;)

Reply:
My main concern with using Approved condition is that in becomes `approved-at-some-point-in-the-past` instead of how we treated it up to now `currently-approved`. It's unclear and error prone IMO.
Annotation counter is IMO better. But I will leave it to you. Keep it as is or switch to annotation on Monday. Or merge it now and then switch to annotation later. I'm fine either way.